daemon/src/db.rs

```diff
@@ -380,6 +430,121 @@ mod tests {
         assert_eq!(events, vec![event1, event2])
     }
 
+    #[tokio::test]
+    async fn given_collaborative_close_confirmed_then_do_not_load_non_final_cfd() {
+        let mut conn = setup_test_db().await;
+
+        let cfd_not_final = insert(dummy_cfd(), &mut conn).await;
+        append_event(lock_confirmed(&cfd_not_final), &mut conn)
+            .await
+            .unwrap();
```

Comment:
Do we need to re-test this in every test?

I think these would be more concise if we had 1 test that ensures a non-final CFD is included and 5 tests that show, which "final" CFDs are excluded.